### PR TITLE
Change the application_choices primary key to an integer

### PR DESCRIPTION
**Pull request by duncanjbrown**
### Context

We used a string for this primary key so that we'd have a single identifier for both vendor API responses (which require a 10-char string) and internal usage. While investigating auditing approaches we realised that having a mixture of string and integer ids inside our app was going to be awkward.

We have decided to see if vendors can live with < 10-char IDs (why not?) and present our numerical ids in the API responses. If this is a problem we could always pad them.

### Changes proposed in this pull request

Create a new vendor_id column which holds the old 10-char string, and create a new id column, which is an auto-incrementing integer, to use as the internal id. Then delete the vendor_id column.

### Link to Trello card

https://trello.com/c/7ubeCZJL/1119-add-audit-trail-for-all-updates-🏈-kickoff

### CODE REVIEW

**Comment by duncanjbrown** [comment]
@tijmenb thank you for the feedback and conversation offline — we now have only one ID 😅

File: spec/presenters/vendor_api/single_application_presenter_spec.rb [hunk 1]
```diff
@@ -20,49 +20,53 @@ def application_choice
...
-        submitted_at: Time.now,
-        updated_at: application_choice.updated_at,
-        withdrawal: nil,
-        work_experiences: [],
       }
     end
 
     it 'returns correct course attributes' do
-      expect(json.dig(:attributes)).to eq expected_attributes
+      expect(json).to eq expected_attributes
```
Comment (RobertM905) [inline spec/presenters/vendor_api/single_application_presenter_spec.rb]: Looks good. 👍 
Could also update the test name here as it's not just course attributes. 
Could be:  `returns correct application attributes`.

Comment (duncanjbrown) [reply spec/presenters/vendor_api/single_application_presenter_spec.rb]: Good catch

Comment (duncanjbrown) [reply spec/presenters/vendor_api/single_application_presenter_spec.rb]: Fixed

**Comment by duncanjbrown** [comment]
Rebased — @RobertM905 are you happy?

**Pull request merged**